src/lib/elements/forms/inputSelect.svelte

```diff
@@ -27,15 +27,29 @@
         error = element.validationMessage;
     };
 
-    $: if (element && required && !value) {
+    const isValid = (value: string | number | boolean) => {
+        if (value === null) return false;
+
+        if (value === undefined) return false;
+
+        if (typeof value !== 'boolean') {
+            if (!value) {
+                return false;
+            }
+        }
+
+        return true;
+    };
+
+    $: if (element && required && !isValid(value)) {
```

Comment:
This can be better expressed, see https://github.com/appwrite/console/pull/406#issuecomment-1511673923

Reply:
Yes, 100%! I just meant that we can use a single `$` sign with `if...else if... else`.

Reply:
updated 👌  Lemme know if this is what you meant

Reply:
It's similar:
```ts
$: if (required && !isNotEmpty(value)) {
    element?.setCustomValidity("This field is required");
  } else {
    element?.setCustomValidity("");
  }

  $: if (isNotEmpty(value)) {
    error = null;
  }
```
With the ?. after element, we don't need to check if element exists.
I kept the `error = null` to separate concerns

Reply:
updated 👌  Thanks for the suggestion :)